prereise/gather/winddata/hrrr/hrrr_api.py

```diff
@@ -0,0 +1,111 @@
+from datetime import timedelta
+
+import requests
+
+from prereise.gather.winddata.hrrr.grib import GribRecordInformation
+from prereise.gather.winddata.hrrr.helpers import (
+    dt_iter,
+    formatted_filename,
+    generate_index_url,
+    get_indices_that_contain_selector,
+)
+
+
+class HrrrApi:
+    """Class to interact with to get and download HRRR data. More information
+    about HRRR data can be found at
+    `this link <https://registry.opendata.aws/noaa-hrrr-pds/>`_
+    and
+    `this link <https://www.nco.ncep.noaa.gov/pmb/products/hrrr/>`_
+
+    :param prereise.gather.winddata.hrrr.downloader.Downloader downloader:
+    """
+
+    # additional sources of data can be found at https://rapidrefresh.noaa.gov/hrrr/
+    # as well as https://github.com/blaylockbk/Herbie/blob/a20d1017ea4c73916cb4436ad797804854109db3/herbie/models/hrrr.py#L58-L63
+    base_url = "https://noaa-hrrr-bdp-pds.s3.amazonaws.com/hrrr.{dt:%Y%m%d}/conus/hrrr.t{dt:%H}z.wrf{product}f{hours_forecasted:02d}.grib2"
```

Comment:
Do we want to allow the user to override this default url somehow, in case the data source changes or they want to use a mirror?

Reply:
yeah, good suggestion